File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        HashMap<String,Integer>hash=new HashMap<>();
        for(String [] arr : clothes){
            if(hash.containsKey(arr[1])){
               hash.put(arr[1],hash.get(arr[1])+1);
            }else{
                hash.put(arr[1],1);
            }
        } 
        int answer = 1;
        for(Integer values: hash.values()){
            answer *= (values+1);
        }
    
        return answer-1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;
map<string,int>m;
int solution(vector<vector<string>> clothes) {
    for(int i=0;i<clothes.size();i++){
        m[clothes[i][1]]++;
    }
    int answer = 1;
    for(auto pair : m)answer*=pair.second+1;
    
    return answer-1;
}